builder/spiffsgen.py

```
@@ -0,0 +1,759 @@
+
+                elif not is_index and flags == SPIFFS_PH_FLAG_USED_FINAL:
+                    # Data page - contains file content
+                    if real_obj_id in files_map:
+                        # Extract content (skip header, no padding on data pages)
+                        content_start = header_size
+                        content = page_data[content_start:content_start + self.build_config.OBJ_DATA_PAGE_CONTENT_LEN]
+                        files_map[real_obj_id]['data_pages'].append((span_ix, content))
+
+        # Extract files to output directory
+        file_count = 0
+        for obj_id, file_info in files_map.items():
```

Ruff (0.14.10)

Loop control variable `obj_id` not used within loop body

(B007)